### PR TITLE
chore: add ipfs http response package

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds the content of https://github.com/ipfs/js-ipfs-http-response to this monorepo to smooth its maintenance burden.

This PR includes all the git contribution history and should be merged instead of squashed.

What's next?
- [ ] add deprecation notice in [github repository](https://github.com/ipfs/js-ipfs-http-response) about the new location of the code -- https://github.com/ipfs/js-ipfs-http-response/pull/107
- [ ] deprecate [github repository](https://github.com/ipfs/js-ipfs-http-response)

### CODE REVIEW

**Comment by achingbrain** [comment]
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /home/runner/work/js-ipfs/js-ipfs/node_modules/ipfs-http-response/package.json
```

This means you're `require`ing an ESM module that hasn't been build with IPJS to be dual CJS/ESM.  Probably need to `lerna link` the dist folder into the relevant `node_modules` folders.

**Comment by vasco-santos** [comment]
yeah, I think I need to use the same think we did in unixfs with publishDirectory

**Comment by achingbrain** [comment]
Maybe open this PR against the branch in #3879 - it's got all the CJS/ESM learnings from `unixfs` and `ipfs-repo` applied so will (hopefully) be less work to get another ESM module merged in.

It's a still a little rough rough the edges though.

**Comment by vasco-santos** [comment]
oh wat, you are changing everything to ESM already 🎉 🎉 🚢 ❤️ ✈️ 

That's rad, yes, I will change the target

**Comment by vasco-santos** [comment]
@achingbrain this will likely get problematic keeping the history of this commit and keeping this branch in sync with #3879 

Should I just wait until that gets merged instead and then update master here?

**Comment by achingbrain** [comment]
That's probably the easiest way - I'm hoping to get it in today or tomorrow.

**Comment by achingbrain** [comment]
#3879 is in so merging master into this should make most of the red disappear.

**Comment by vasco-santos** [comment]
Thanks! 🎉 I will look into it

File: packages/ipfs-http-response/test/resolver.spec.js
Comment (achingbrain) [inline packages/ipfs-http-response/test/resolver.spec.js:7]:
```suggestion
import { create } from 'ipfs-core'
```

**Comment by vasco-santos** [comment]
@achingbrain this is ready.

The commits should not be squashed, in order to keep repo history and contributors

**Review by achingbrain — APPROVED** [review]

**Pull request merged**
